packages/kbn-apm-synthtrace-client/src/lib/apm/event.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ApmFields } from './apm_fields';
+import { Serializable } from '../serializable';
+
+export class Event extends Serializable<ApmFields> {
+  constructor(fields: ApmFields) {
+    super({
+      ...fields,
+      'event.kind': 'event',
+
+      // Real log events don't have the "processor.event" attribute. See: https://github.com/elastic/apm-server/pull/11494
+      // This is a hack needed for the tests in "mobile/mobile_location_stats.spec.ts" to pass since the "SynthtraceEsClient.index" function
+      // inside "packages/kbn-apm-synthtrace/src/lib/shared/base_client.ts" requires this "processor.event" attr to be available,
+      // otherwise it'd throw this error: "Could not determine operation: _index and _action not defined in document".
+      'processor.event': 'error',
```

Comment:
I'm a bit concerned with this hack, introducing a new field in "synthrace" that isn't present in "apm-server" could easily result in bugs. For instance, any modification to the query in order to filter data based on the "processor.event" field might initially pass the API test but could subsequently cause issues.

it seems that the `processor.event` attribute is needed to determine the index name . Which is defined here https://github.com/elastic/kibana/blob/main/packages/kbn-apm-synthtrace/src/lib/apm/client/apm_synthtrace_es_client/get_routing_transform.ts#L20

what do you think of extending the function to accept log events that don't have this attribute but still return the corresponding index name.

Reply:
I also wouldn't like to go with the hack tbh, though I wasn't sure where could I fix that issue properly until now 😅 - I'll have a look at the file you mentioned, cheers.

Reply:
Thanks, I've updated `get_routing_transform.ts` to cover the event use case and removed the hack 👍